### PR TITLE
GDB-14452: Refactor GeoSPARQL parsing and enrich geo features support

### PR DESCRIPTION
**Pull request by yordanalexandrov**
## What
Refactored the GeoSPARQL service to modularize SRID definitions and parsers. Enhanced the geo plugin to support additional geometric datatypes with extensive test coverage for edge cases and malformed inputs.

## Why
Improves code maintainability through a singleton projection manager, enhances geometry parsing robustness, and ensures broader support for GeoSPARQL literal formats and GeoURI edge cases.

## How
- Replaced inline SRID definitions with a centralized `SridProjDefinitions` utility and renamed the class for naming consistency.
- Introduced `Proj4Instance` as a singleton manager to handle lazy registration of EPSG definitions and prevent circular dependencies.
- Enhanced `Parsers` utility with improved robustness:
    - Added support for H3/DGGS literals without URI prefixes and both comma/whitespace separated cell lists.
    - Improved GeoURI parsing to handle optional parameters and case-insensitive scheme matching.
    - Implemented strict coordinate validation and added comprehensive warning logs for parsing failures.
    - Refined `extractUriPrefix` logic to avoid false positives with malformed literal prefixes.
- Improved `gml-parser` with stricter validation for finite number coordinates in GML 2 and 3 formats.
- Added Cypress fixtures and tests to verify rendering of complex scenarios, including multi-column geometries, invalid WKT/GeoJSON, and various GeoURI formats.

### CODE REVIEW

File: ontotext-yasgui-web-component/package.json [hunk 1]
```diff
@@ -43,6 +46,7 @@
     "@types/leaflet": "^1.9.21",
     "@typescript-eslint/eslint-plugin": "^5.46.1",
     "@typescript-eslint/parser": "^5.46.1",
+    "@xmldom/xmldom": "^0.9.10",
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/package.json:49]: It is needed by an external library

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+   * @param input - The string to extract the URI prefix from.
+   * @returns An object with the extracted `uri` and the remaining `value`.
+   */
+  private static extractUriPrefix(input: string): { uri: string; value: string } {
+    if (input.startsWith('<')) {
+      const closingIndex = input.indexOf('>');
+      return {
+        uri: input.substring(1, closingIndex),
+        value: input.substring(closingIndex + 1).trim()
+      };
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts]: added

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+   * @param dggs - A DGGS literal string, optionally prefixed with a URI in angle brackets.
+   * @returns A GeoJSON Geometry object, or `null` if the format is not supported.
+   */
+  static parseDGGS(dggs: string): Geometry | null {
+    const {uri: dggsUri, value} = Parsers.extractUriPrefix(dggs.trim());
+    if (dggsUri.startsWith('https://h3geo.org/res/')) {
+      return Parsers.parseH3(value);
+    }
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts:97]: added

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    };
+    // This regex matches any of the keys in our map
+    const normalized = dggs.replace(/CELLLIST|CELL|\(|\)|'/g, (matched) => map[matched]);
+    const dggsdict = JSON.parse(normalized);
+    if (isPoint) {
+      const decoded = cellToLatLng(dggsdict[0]);
+      return {'type': 'Point', 'coordinates': [decoded[1], decoded[0]]};
+    }
+    const result = cellsToMultiPolygon(dggsdict, true);
+    return {'type': 'MultiPolygon', 'coordinates': result};
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts]: changed

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    }
+    return {uri: '', value: input};
+  }
+
+  /**
+   * Parses an H3 DGGS cell or cell list string into a GeoJSON Geometry.
+   *
+   * A single `CELL` is decoded into a GeoJSON Point; a `CELLLIST` is decoded into a MultiPolygon.
+   *
+   * @param dggs - An H3 DGGS string using `CELL('id')` or `CELLLIST('id1' 'id2' ...)` syntax.
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts]: changed

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+        return null;
+      }
+    }
+    if (geocodeuri === 'http://opengis.net/ont/geocode/GeoURI') {
+      const splitted = value.replace(/geo:/g, '').split(',');
+      if (splitted.length === 2) {
+        return {'type': 'Point', 'coordinates': [Number.parseFloat(splitted[1]), Number.parseFloat(splitted[0])]};
+      } else if (splitted.length === 3) {
+        return {'type': 'Point', 'coordinates': [Number.parseFloat(splitted[1]), Number.parseFloat(splitted[0]), Number.parseFloat(splitted[2])]};
+      }
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/parsers.ts]: changed

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/gml-parser.ts [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  return Array.from(el.children).filter(c => localName(c) === name);
+}
+
+/**
+ * Parse a coordinate string like "1.0,2.0" or "1.0 2.0" or "1.0,2.0,3.0".
+ * GML 3 uses space-separated tuples; GML 2 uses comma-separated within a tuple.
+ */
+function parseCoordTuple(s: string): number[] {
+  return s.trim().split(/[\s,]+/).map(Number);
+}
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/utils/gml-parser.ts:75]: changed

File: ontotext-yasgui-web-component/src/plugins/yasr/geo/services/geo-sparql-service.ts [hunk 8]
```diff
@@ -1,8 +1,8 @@
-import { Geometry } from 'geojson';
-import { wktToGeoJSON } from 'betterknown';
 import proj4 from 'proj4';
-import { BindingValue } from '../../../../models/yasgui/parser';
-import {ObjectUtil} from '../../../../services/utils/object-util';
+import {BindingValue} from '../../../../models/yasgui/parser';
+import {SRIDProjDefinitions} from '../utils/SRIDProjDefinitions';
+import {Parsers} from '../utils/parsers';
+import {Geometry} from 'geojson';
```
[the remark this answers is not in the document]
Comment (yordanalexandrov) [reply ontotext-yasgui-web-component/src/plugins/yasr/geo/services/geo-sparql-service.ts]: changed

**Review by boyan-tonchev — APPROVED** [review]
